gatsby-node.js

```diff
@@ -5,3 +5,59 @@
  */
 
 // You can delete this file if you're not using it
+exports.createSchemaCustomization = ({ actions }) => {
+  const { createTypes } = actions
+  const typeDefs = `
+    type MarkdownRemarkFrontmatterSection implements Node {
+      section_2: Section
+    }
+    type Section implements Node {
+      title: String
+      introduction: String
+    }
+  `
+  createTypes(typeDefs)
```

Comment:
This is one way we can make sections optional. We still need to list each `section_X` but we can set a hard limit on them. 

https://www.gatsbyjs.org/docs/schema-customization/#opting-out-of-type-inference